src/back/TicketManager/src/PaymentService/TicketManager.PaymentService.Services/Endpoints/Payments/CancelPaymentEndpoint.cs

```diff
@@ -0,0 +1,46 @@
+using FastEndpoints;
+using TicketManager.PaymentService.Contracts.Payments;
+using TicketManager.PaymentService.Domain.Payments;
+using TicketManager.PaymentService.Services.DataAccess.Repositories;
+
+namespace TicketManager.PaymentService.Services.Endpoints.Payments;
+
+public class CancelPaymentEndpoint: Endpoint<CancelPaymentRequest, EmptyResponse>
+{
+    private readonly Repository<Payment, Guid> payments;
+
+    public CancelPaymentEndpoint(Repository<Payment,Guid> payments)
+    {
+        this.payments = payments;
+    }
+
+    public override void Configure()
+    {
+        Post("/payment/cancel");
+        AllowAnonymous();
+    }
+
+    public override async Task HandleAsync(CancelPaymentRequest req, CancellationToken ct)
+    {
+        try
```

Comment:
So instead of catching exceptions in handler, create validator that prevents running handler in situations that it shouldn't be run.

Reply:
Then you will have clear handler, that only realize its logic and not mix logix with validation.

Reply:
As I understand I shouldn't put status check in the validator because it does have logic? (Deferred expiration). Maybe I should put "Payment already decided" check in validation? Confirmation/canellation would still be in try/catch because expiration has more complex logic.

Reply:
No, all should be in validator. Expiration is not that complex. You have to check just 2 values status and date.

Reply:
Addressed in 7f75126c.